### PR TITLE
fix(web): stabilize loading after organization setup

### PR DESCRIPTION
**Pull request by GuiBibeau**
After organization creation, loading jumped from a standalone Home skeleton to generic workspace cards and then a differently sized dashboard shell. The ready redirect also reloaded the document and restarted Clerk.

This keeps navigation inside Next.js and shares the destination skeleton and shell geometry across preparation, scope reconciliation, and client auth. Home's placeholder follows its onboarding/dismissal state, and hiding the balance card no longer leaves an empty animated wrapper. The standard Home skeleton remains until organization state is known; confirmed new sandbox organizations then use the compact card. The explicit preparation message stays visible without moving the cards. Polling deadlines, explicit retries, and scope isolation are preserved. No API changes.

Validation: web typecheck, scoped Biome, module-boundary generation/check, and diff checks passed. The full web unit run passed 2,202/2,203 tests; the Earn dropdown test passed on targeted rerun without changes. All loading/onboarding tests passed. Local component QA covered 1440×900 light/dark and 390×844 dark with no horizontal overflow; the compact loading and settled empty card have matching bounds. These screenshots use production components with stubbed workspace/data, not a live signup/webhook replay.

Desktop dark

<details><summary>Light and mobile screenshots</summary>

Desktop light
Mobile dark

Preparation status

</details>

### CODE REVIEW

**Comment by GuiBibeau** [comment]
@greptileai review

File: apps/sdp-web/src/components/dashboard-page-loading.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  WalletSetupSkeleton,
+  WalletsOverviewSkeleton,
+} from "@/app/dashboard/wallets/wallet-route-skeletons";
+import type { DashboardLoadingRoute } from "@/lib/dashboard-navigation-loading";
+
+function ApiKeyNewLoading() {
+  return <ApiKeyAuthoringSkeleton route="api-key-new" />;
+}
+
+function ApiKeyEditLoading() {
```
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: These two-line route adapters were moved unchanged from DashboardShell into the shared loading resolver. Keeping them beside the route mapping makes preparation, auth, and workspace switching use the same skeleton without adding separate files for prop presets. The actual skeleton components remain in their route modules; no component state or behavior was added here.

File: apps/sdp-web/src/components/dashboard-page-loading.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  return <ApiKeyAuthoringSkeleton route="api-key-edit" />;
+}
+
+function AllowlistLoading() {
```
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: These two-line route adapters were moved unchanged from DashboardShell into the shared loading resolver. Keeping them beside the route mapping makes preparation, auth, and workspace switching use the same skeleton without adding separate files for prop presets. The actual skeleton components remain in their route modules; no component state or behavior was added here.

File: apps/sdp-web/src/components/dashboard-page-loading.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  return <CompactOperationsCardSkeleton route="allowlist" />;
+}
+
+interface PageLoadingProps {
+  assetProfilesEnabled?: boolean;
+}
+
+function CounterpartyDirectoryLoading() {
```
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: These two-line route adapters were moved unchanged from DashboardShell into the shared loading resolver. Keeping them beside the route mapping makes preparation, auth, and workspace switching use the same skeleton without adding separate files for prop presets. The actual skeleton components remain in their route modules; no component state or behavior was added here.

File: apps/sdp-web/src/components/dashboard-page-loading.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  return <CounterpartyMenuLoading overview="counterparty-directory" />;
+}
+
+function PaymentRequestsLoading() {
```
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: These two-line route adapters were moved unchanged from DashboardShell into the shared loading resolver. Keeping them beside the route mapping makes preparation, auth, and workspace switching use the same skeleton without adding separate files for prop presets. The actual skeleton components remain in their route modules; no component state or behavior was added here.

File: apps/sdp-web/src/components/fullscreen-loading-indicator.tsx [hunk 5]
```diff
@@ -52,79 +39,78 @@ export function FullscreenLoadingIndicator({
...
         <div
-          aria-hidden="true"
-          className={`mx-auto w-full ${contentWidthClass} space-y-6 px-3 py-5 md:p-6`}
+          role="status"
+          aria-live="polite"
+          className={
+            recoveryAction
+              ? "sticky bottom-20 mx-auto flex max-w-xl items-center justify-between gap-4 rounded-xl border border-border-default bg-surface-raised p-4 shadow-sm md:bottom-6"
+              : "sr-only"
+          }
```
[the remark this answers is not in the document]
Comment (GuiBibeau) [reply apps/sdp-web/src/components/fullscreen-loading-indicator.tsx:106]: Fixed in 7eb6e4303: an explicit preparation status stays visible even before a recovery action is offered. The notice sits after the content and sticks within the viewport, so the cards retain their position. Added a regression assertion and desktop/mobile preparation screenshots.

**Comment by GuiBibeau** [comment]
@greptileai review

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 1 above)
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: Duplicate advisory from the new CI run. The adapters are unchanged; rationale is recorded at https://github.com/solana-foundation/solana-developer-platform/pull/1784#discussion_r3990612132. Keeping these small prop presets with their shared route resolver avoids unnecessary files and preserves the existing skeleton implementations.

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 2 above)
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: Duplicate advisory from the new CI run. The adapters are unchanged; rationale is recorded at https://github.com/solana-foundation/solana-developer-platform/pull/1784#discussion_r3990612132. Keeping these small prop presets with their shared route resolver avoids unnecessary files and preserves the existing skeleton implementations.

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 3 above)
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: Duplicate advisory from the new CI run. The adapters are unchanged; rationale is recorded at https://github.com/solana-foundation/solana-developer-platform/pull/1784#discussion_r3990612132. Keeping these small prop presets with their shared route resolver avoids unnecessary files and preserves the existing skeleton implementations.

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 4 above)
Comment (GuiBibeau) [inline apps/sdp-web/src/components/dashboard-page-loading.ts]: Duplicate advisory from the new CI run. The adapters are unchanged; rationale is recorded at https://github.com/solana-foundation/solana-developer-platform/pull/1784#discussion_r3990612132. Keeping these small prop presets with their shared route resolver avoids unnecessary files and preserves the existing skeleton implementations.

**Comment by GuiBibeau** [comment]
Ready for QA on 7eb6e4303. Greptile is 5/5, all nine review threads are resolved, and all 47 applicable checks passed (six skipped).

Preview: https://sdp-web-git-codex-organization-loading-87569c-solana-foundation.vercel.app/dashboard

Please verify a fresh organization's setup-to-Home transition, the visible preparation/retry message during slow sync, and switching between existing organizations. The loading frame should stay aligned, Home should use its compact new-organization layout once state is known, and dismissed onboarding should retain the normal balance layout. Local desktop/mobile screenshots and validation details are in the PR body and docs/qa/organization-loading.

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (GuiBibeau) [reply apps/sdp-web/src/components/dashboard-page-loading.ts]: Fixed in 40094f66f. The resolver imports the existing API key creation route loading component directly. All five duplicate adapters were removed, leaving a plain TypeScript resolver with no component declarations. Typecheck and 35 loading tests pass.

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuiBibeau) [reply apps/sdp-web/src/components/dashboard-page-loading.ts]: Fixed in 40094f66f. The resolver imports the existing API key editing route loading component directly. All five duplicate adapters were removed, leaving a plain TypeScript resolver with no component declarations. Typecheck and 35 loading tests pass.

**Comment by GuiBibeau** [comment]
@greptileai review

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (GuiBibeau) [reply apps/sdp-web/src/components/dashboard-page-loading.ts]: Fixed in 40094f66f. The resolver imports the existing counterparty directory route loading component directly. All five duplicate adapters were removed, leaving a plain TypeScript resolver with no component declarations. Typecheck and 35 loading tests pass.

File: apps/sdp-web/src/components/dashboard-page-loading.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuiBibeau) [reply apps/sdp-web/src/components/dashboard-page-loading.ts]: Fixed in 40094f66f. The resolver imports the existing payment requests route loading component directly. All five duplicate adapters were removed, leaving a plain TypeScript resolver with no component declarations. Typecheck and 35 loading tests pass.

**Comment by GuiBibeau** [comment]
React Doctor feedback is fixed in 40094f66f: the resolver reuses the existing route loading components, removing five duplicate component declarations and 21 net lines. The UI output is preserved.

Ready for QA again: React Doctor passes, Greptile is 5/5 on the latest commit, all review threads are resolved, and all 46 applicable checks passed (six skipped). Local validation also passed typecheck, 35 targeted tests, Biome, module boundaries, and a focused React Doctor scan with no issues.

**Review by multipletwigs — APPROVED** [review]
